domains/runtime/evm/src/lib.rs

```diff
@@ -1121,43 +1121,40 @@ fn check_transaction_and_do_pre_dispatch_inner(
                     extra.8,
                 );
 
-                let nonce_val =
-                    pallet_evm_tracker::check_nonce::Val::CheckNonce((account_id, extra.5 .0));
-                let (_, weight_val) =
-                    frame_system::CheckWeight::<Runtime>::do_validate(&dispatch_info, encoded_len)?;
+                let origin = RuntimeOrigin::none();
+                <CustomSignedExtra as DispatchTransaction<RuntimeCall>>::validate_and_prepare(
+                    custom_extra,
+                    origin,
+                    &xt.function,
+                    &dispatch_info,
+                    encoded_len,
+                    extension_version,
+                )
+                .map(|_| ())
+            }
+            ExtrinsicFormat::Signed(account_id, extra) => {
+                let custom_extra: CustomSignedExtra = (
+                    extra.0,
+                    extra.1,
+                    extra.2,
+                    extra.3,
+                    extra.4,
+                    pallet_evm_tracker::CheckNonce::from(extra.5 .0),
+                    extra.6,
+                    extra.7.clone(),
+                    extra.8,
+                );
 
                 let origin = RuntimeOrigin::signed(account_id);
-
-                let implication = TxBaseImplication(xt.function.clone());
-                let (_, transaction_payment_val, origin) =
-                    pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::validate(
-                        &extra.7,
-                        origin,
-                        &xt.function,
-                        &dispatch_info,
-                        encoded_len,
-                        (),
-                        &implication,
-                        TransactionSource::External,
-                    )?;
-
-                <CustomSignedExtra as TransactionExtension<RuntimeCall>>::prepare(
+                <CustomSignedExtra as DispatchTransaction<RuntimeCall>>::validate_and_prepare(
                     custom_extra,
-                    (
-                        (),
-                        (),
-                        (),
-                        (),
-                        (),
-                        nonce_val,
-                        weight_val,
-                        transaction_payment_val,
-                        (),
-                    ),
-                    &origin,
+                    origin,
                     &xt.function,
                     &dispatch_info,
                     encoded_len,
+                    // default extension version define here -
+                    // https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/runtime/src/generic/checked_extrinsic.rs#L37
+                    0,
```

Comment:
Can we port that const value to our codebase? so we don't need to copy paste these lines where place.